### PR TITLE
use .to_string() instead of format!() macro to create strings

### PR DESCRIPTION
**Pull request by matthiaskrgr**
handles what is left after https://github.com/rust-lang/rust/pull/69541

### CODE REVIEW

File: src/librustc_lint/types.rs [hunk 1]
```diff
@@ -83,8 +83,8 @@ fn lint_overflowing_range_endpoint<'a, 'tcx>(
                     // We need to preserve the literal's suffix,
                     // as it may determine typing information.
                     let suffix = match lit.node {
-                        LitKind::Int(_, LitIntType::Signed(s)) => format!("{}", s.name_str()),
-                        LitKind::Int(_, LitIntType::Unsigned(s)) => format!("{}", s.name_str()),
+                        LitKind::Int(_, LitIntType::Signed(s)) => s.name_str().to_owned(),
+                        LitKind::Int(_, LitIntType::Unsigned(s)) => s.name_str().to_owned(),
```
Comment (nagisa) [inline src/librustc_lint/types.rs]: Any reason these are `to_owned`?

File: src/librustc_parse/parser/pat.rs [hunk 2]
```diff
@@ -216,7 +216,7 @@ impl<'a> Parser<'a> {
             .span_suggestion(
                 seq_span,
                 "...or a vertical bar to match on multiple alternatives",
-                format!("{}", seq_snippet.replace(",", " |")),
+                seq_snippet.replace(",", " |").to_string(),
```
Comment (nagisa) [inline src/librustc_parse/parser/pat.rs]: `replace` already returns `String`.

File: src/librustc_lint/types.rs (hunk 1 above)
Comment (matthiaskrgr) [reply src/librustc_lint/types.rs]: the line below was already using `to_owned()` so I thought I would go with it. :shrug:

Comment (matthiaskrgr) [reply src/librustc_lint/types.rs]: I switched all of them to to_string() now.

**Pull request merged**